### PR TITLE
Improve wall following

### PR DESCRIPTION
**Pull request by knmcguire**
For this PR ive introduced the wall following demo: https://github.com/bitcraze/crazyflie-lib-python/pull/390

There were comments made by @krichardsson  there that I've addressed +  more documentation and clarification

### CODE REVIEW

File: examples/multiranger/multiranger_wall_following.py [hunk 1]
```diff
@@ -80,54 +85,45 @@
...
-                        # get front range in milimeters
-                        front_range = multiranger.front
-                        if front_range is None:
-                            front_range = 999
-
-                        # get top range in milimeters
-                        top_range = multiranger.up
-                        if top_range is None:
-                            top_range = 999
+                        # get front range in meters
```
Comment (krichardsson) [inline examples/multiranger/multiranger_wall_following.py:102]: Much more readable, nice!

File: examples/multiranger/wall_following/wall_following.py [hunk 2]
```diff
@@ -94,192 +129,250 @@ def wrap_to_pi(self, number):
...
 
-    def command_align_corner(self, ref_rate, range, wanted_distance_from_corner):
-        if range > wanted_distance_from_corner + 0.3:
-            cmd_ang_w = self.direction * ref_rate
-            cmd_vel_y = 0
+        side_range and wanted_distance_from_corner is defined in m
+        reference_rate and rate_yaw is defined in rad/s
+        velocity_x is defined in m/s
+        """
+        if side_range > wanted_distance_from_corner + 0.3:
```
Comment (krichardsson) [inline examples/multiranger/wall_following/wall_following.py:153]: Is this 0.3 the same as `range_lost_threshold`? If not, maybe use a constant or variable?

File: examples/multiranger/wall_following/wall_following.py [hunk 3]
```diff
@@ -94,192 +129,250 @@ def wrap_to_pi(self, number):
...
+            case self.StateWallFollowing.TURN_TO_FIND_WALL:
+                side_range_check = side_range < (self.reference_distance_from_wall /
                                                  math.cos(0.78) + self.ranger_value_buffer)
-                front_range_check = front_range < (self.ref_distance_from_wall /
+                front_range_check = front_range < (self.reference_distance_from_wall /
                                                    math.cos(0.78) + self.ranger_value_buffer)
                 if side_range_check and front_range_check:
                     self.prev_heading = current_heading
-                    self.wall_angle = self.direction * \
+                    self.wall_angle = self.wall_following_direction_value * \
```
Comment (krichardsson) [inline examples/multiranger/wall_following/wall_following.py:274]: math.pi / 2 would perhaps be clearer than 1.57?

File: examples/multiranger/wall_following/wall_following.py [hunk 4]
```diff
@@ -94,192 +129,250 @@ def wrap_to_pi(self, number):
...
+        match self.state:
+            case self.StateWallFollowing.FORWARD:
+                if front_range < self.reference_distance_from_wall + self.ranger_value_buffer:
+                    self.state = self.state_transition(self.StateWallFollowing.TURN_TO_FIND_WALL)
+            case self.StateWallFollowing.HOVER:
                 print('hover')
-            case self.StateWF.TURN_TO_FIND_WALL:
-                side_range_check = side_range < (self.ref_distance_from_wall /
+            case self.StateWallFollowing.TURN_TO_FIND_WALL:
+                side_range_check = side_range < (self.reference_distance_from_wall /
```
Comment (krichardsson) [inline examples/multiranger/wall_following/wall_following.py:268]: math.pi / 4 would perhaps be clearer than 0.78?

File: examples/multiranger/multiranger_wall_following.py [hunk 5]
```diff
@@ -80,54 +85,45 @@
...
                         velocity_y = 0.0
                         yaw_rate = 0.0
-                        state_wf = WallFollowing.StateWF.HOVER
+                        state_wf = WallFollowing.StateWallFollowing.HOVER
 
                         # Get Yaw
                         log_entry = logger.next()
                         data = log_entry[1]
                         actual_yaw = data['stabilizer.yaw']
                         actual_yaw_rad = actual_yaw * 3.1415 / 180
```
Comment (krichardsson) [inline examples/multiranger/multiranger_wall_following.py:100]: You could use math.radians() here for clarity

**Review by krichardsson — APPROVED** [review]
Much nicer and easier to read!

I found a few magic numbers (comments in the code) if you want to fix them, but nothing big.

Good job!

File: examples/multiranger/wall_following/wall_following.py (hunk 2 above)
Comment (knmcguire) [reply examples/multiranger/wall_following/wall_following.py:153]: Yeah I think that is probably supposed to be that variable. Changing it!

**Comment by knmcguire** [comment]
Thanks for reviewing! And good you found those extra magic numbers... wonder if there is some kind of CI check for this?

I've made some changes per your suggestion and will merge it now.

**Pull request merged**